### ISSUE TITLE
Load testing - FastAPI much slower than Flask?

### ISSUE THREAD

**Issue by MaxHoefl** [tiangolo/fastapi#5908]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from uuid import uuid4

import uvicorn
from fastapi import FastAPI
from time import sleep
import os
from loguru import logger

app = FastAPI()


@app.get("/health")
def health():
    return "Ok"


@app.get("/load")
async def run_load_test():
    request_id = str(uuid4())
    logger.info(f"Starting request {request_id}")
    sleep(2)
    logger.info(f"Finished request {request_id}")
    return "Ok"


if __name__ == "__main__":
    port = int(os.environ.get('PORT', 6000))
    uvicorn.run("main:app", host="0.0.0.0", port=port, log_level="debug")
```

### Description

First of all, huge fan, big thanks for this awesome library ❤️ 

* I am comparing FastAPI vs Flask "under load" using [Apache Jmeter](https://jmeter.apache.org/). Jmeter allows me to issue 100 parallel requests to the `/load` endpoint of the above FastAPI app. It does so by generating 100 threads within 0.01 seconds that then issue the GET request.
* Jmeter times the roundtrip latency from each thread and generates a report
* The Jmeter report for both the FastAPI and the Flask app can be seen below:
      FastAPI took on average `3727.85ms` per request
      Flask took on average `2072.45ms` per request
* Somehow I expected FastAPI to be... well "faster". Please see the Github repo [here](https://github.com/MaxHoefl/demo-api-perf) to reproduce the findings

### Operating System

macOS

### Operating System Details

OS: macOS Monterey  (version: 12.3 (21E230))
Model: MacBook Pro (14-inch, 2021)
Chip: Apple M1 Pro
Memory: 32 GB

### FastAPI Version

0.89.1

### Python Version

Python 3.9.13

### Additional Context

Here is a minimal example to reproduce the findings: https://github.com/MaxHoefl/demo-api-perf

**Comment by fmazan** [comment]
I think the problem is that you are calling synchronous `time.sleep(2)` inside the `async` method. This causes fastapi to make wrong assumptions and it does not offload the task to the threadpool. Try changing it to `await asyncio.sleep(2)` or making the path operation method synchronous by removing `async`.

More reading [in this article](https://github.com/zhanymkanov/fastapi-best-practices#7-dont-make-your-routes-async-if-you-have-only-blocking-io-operations).

**Comment by MaxHoefl** [comment]
@fmazan yes, with you suggestion, FastAPI takes on average `2111.15ms` per request vs `2144.22ms` in Flask
